src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { useAppSelector } from '../../app/hooks';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+export const TodoModal = () => {
+  const dispatch = useDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
 
-      <Loader />
+  const { id, title, completed, userId } = todo || {};
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    if (userId) {
+      setIsLoading(true);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+      getUser(userId)
+        .then(setUser)
+        .catch(error => {
+          throw new Error(error);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  }, [userId]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  const handleCloseTodo = () => {
+    dispatch(setCurrentTodo(null));
```

Comment:
You are directly dispatching the `setCurrentTodo` action with null as payload. This could lead to potential bugs in your application if other parts of your code expect the current todo to always be an object. Consider adding a `clearCurrentTodo` action that doesn't require a payload.